sdk/eventhub/Azure.Messaging.EventHubs/tests/EventHubConsumerClient/EventHubConsumerClientLiveTests.cs

```diff
@@ -1111,7 +1112,126 @@ await foreach (var partitionEvent in consumer.ReadEventsAsync(DefaultReadOptions
         /// </summary>
         ///
         [Test]
-        public async Task ConsumerCannotReadWhenClosed()
+        public async Task ConsumerCanReadUsingAnIdentityCredential()
+        {
+            await using (EventHubScope scope = await EventHubScope.CreateAsync(1))
+            {
+                var credential = new ClientSecretCredential(TestEnvironment.EventHubsTenant, TestEnvironment.EventHubsClient, TestEnvironment.EventHubsSecret);
+
+                await using (var producer = new EventHubProducerClient(TestEnvironment.FullyQualifiedNamespace, scope.EventHubName, credential))
+                await using (var consumer = new EventHubConsumerClient(EventHubConsumerClient.DefaultConsumerGroupName, TestEnvironment.FullyQualifiedNamespace, scope.EventHubName, credential))
+                {
+                    var partition = (await consumer.GetPartitionIdsAsync()).First();
+                    var expectedEventsCount = 10;
+
+                    for (int i = 0; i < expectedEventsCount; i++)
+                    {
+                        await producer.SendAsync(new EventData(new byte[1]), new SendEventOptions { PartitionId = partition });
+                    }
+
+                    // Read the events.
+
+                    var consecutiveEmpties = 0;
+                    var receivedEvents = new List<EventData>();
+
+                    await foreach (PartitionEvent partitionEvent in consumer.ReadEventsFromPartitionAsync(partition, EventPosition.Earliest, DefaultReadOptions))
+                    {
+                        if (partitionEvent.Data != null)
+                        {
+                            receivedEvents.Add(partitionEvent.Data);
+                            consecutiveEmpties = 0;
+                        }
+                        else if (++consecutiveEmpties >= 5)
+                        {
+                            break;
+                        }
+                    }
+
+                    Assert.That(receivedEvents.Count, Is.EqualTo(expectedEventsCount), $"The number of received events should be { expectedEventsCount }.");
```

Comment:
Should we check the cancellation token and `consecutiveEmpties` here as well?

Reply:
We could, but the purpose of the test is to validate that we could perform basic scenarios using an identity credential.   The proof that we can read properly across all partitions is tested elsewhere; this just ensures that we can open multiple links and ask to read with a principal identity and we get **something** back.   We're just asserting the count to be sure that we got things from all partitions in a lazy way rather than writing another few lines to check each partition individually.